Sources/TuistKit/Utils/ManifestGraphLoader.swift

```diff
@@ -144,6 +148,11 @@ public final class ManifestGraphLoader: ManifestGraphLoading {
 
         // Check circular dependencies
         try graphLoaderLinter.lintWorkspace(workspace: workspaceModels, projects: projectsModels)
+        let filesLintingIssues = filesLinter.lint(
+            project: allManifests,
+            convertedProjects: projectsModels
+        )
+        filesLintingIssues.printWarningsIfNeeded()
```

Comment:
Why is this linted here and not along with all the other linters we have [here](https://github.com/tuist/tuist/tree/main/Sources/TuistGenerator/Linter)? This could probably be added as a [TargetLinter](https://github.com/tuist/tuist/blob/main/Sources/TuistGenerator/Linter/TargetLinter.swift). I don't really see a reason why this needs to be linted during the load phase.

Reply:
Yeah, so i tried to looking into that, however it seems that we cant really put it there since we require the `allManifests` and that's being fetched within the same function. and the linter itself compares the paths between, the generated targets from the `allManifests` to the actual `allManifests`

Reply:
Or do you think there is a better option there @fortmarek?

Reply:
Considering the need "I need linters to know whether a file has been derived from a glob", you need:
1. A way to express that
2. Use that information from the linters that @fortmarek pointed out.

We already [have a module](https://github.com/tuist/tuist/blob/main/Sources/TuistGraph/Models/FileElement.swift#L4) that represents a path. You can extend it to include information about whether it came from a glob:

```swift
public enum FileElement: Equatable, Hashable, Codable {
    case file(path: AbsolutePath, metadata: Set<FileElementMetadata>)
    case folderReference(path: AbsolutePath, metadata: Set<FileElementMetadata>)
}

// Somewhere in the mapping logic.
FileElement.path("....", metadata: [.fromGlob])
```

Reply:
Okay nice then, I'll take a look at it and see if that's going to work as expected

Reply:
So I looked at the code and how its structure, and it seems that the better approach would be simply adding the paths into for example:
```
TuistGraph.Target.sourcesPaths: [FileElement.path("....", metadata: [.fromGlob]), FileElement.path("....", metadata: [.file])]
```
and this will also be translated to resourcesPaths, and copyFilesPaths. keeping the original sources, resources, and copyFiles, since this would be the main change that would not affect the codebase as much.

```swift
public struct Target: Equatable, Hashable, Comparable, Codable {
   ...
    public var sources: [SourceFile]
    public var sourcesPath: [FileElement]
    public var resources: [ResourceFileElement]
    public var resourcesPath: [FileElement]
    public var copyFiles: [CopyFilesAction]
    public var copyFilesPath: [FileElement]
    ...
}
```

Reply:
I've been wondering, do we need the extra metadata? If we decide that we won't show any warnings when no files match a glob but we still want to show a warning for individual files, can't we only lint the individual files exist? As we don't add the parent glob directory if the glob doesn't match anything.

Reply:
> can't we only lint the individual files exist?

That's the idea, but how do you know at the linter level if a file or not was hardcoded by the user or resolved from a glob without passing that information when the models are mapped from `ProjectDescription` to `TuistGraph`?

Reply:
When we resolve a glob, we then pass on only files that the file system can find. How would a non-existing file from a glob be added as a regular file?

Reply:
I'll resume the work on this weekend, or the weekend after. I've been super busy lately.

Reply:
Hey, If possible, I would love to get some feedback regarding the approach that was taken before adding tests and basically moving forward with this PR.